### PR TITLE
Collect query string by default and obfuscate

### PR DESCRIPTION
**Pull request by Anilm3**
### Description

Collect the query string by default as part of `http.url` and add an obfuscation step using a predefined regex. Some considerations:
- The regex is configurable with the variable `DD_OBFUSCATION_QUERY_STRING_REGEXP`.
- The obfuscation is only done when `DD_TRACE_HTTP_URL_QUERY_PARAM_ALLOWED` is set to `*` (which is now the default value as well).

### Readiness checklist
- [ ] (only for Members) Changelog has been added to the release document.
- [ ] Tests added for this feature/bug.

### Reviewer checklist
- [ ] Appropriate labels assigned.
- [ ] Milestone is set.
- [ ] Changelog has been added to the release document. For community contributors the reviewer is in charge of this task.

### CODE REVIEW

File: tests/ext/root_span_url_with_query_params.phpt [hunk 1]
```diff
@@ -0,0 +1,28 @@
+--TEST--
```
Comment (Anilm3) [inline tests/ext/root_span_url_with_query_params.phpt:1]: This test was just renamed, but another test was created with the same name this one used to have so it appears as new instead...

File: ext/php7/serializer.c [hunk 2]
```diff
@@ -555,7 +556,7 @@ void ddtrace_set_root_span_properties(ddtrace_span_t *span) {
                 if (query_str) {
                     query_string =
                         zai_filter_query_string((zai_string_view){.len = strlen(query_str), .ptr = query_str},
-                                                get_DD_TRACE_RESOURCE_URI_QUERY_PARAM_ALLOWED());
```
Comment (bwoebi) [inline ext/php7/serializer.c:558]: This does not look right to me. With `DD_TRACE_URL_AS_RESOURCE_NAMES_ENABLED=true` the query string is sent unmodified by default to datadog, given the new `*` default. I think at this place we need the regex filter as well.

Comment (Anilm3) [reply ext/php7/serializer.c:558]: Well, the default for `DD_TRACE_RESOURCE_URI_QUERY_PARAM_ALLOWED` is still an empty array, but I guess it also makes sense to add the regex here for the `*` case, so I'll do that.

File: ext/php7/serializer.c [hunk 3]
```diff
@@ -10,6 +10,7 @@
 #include <stdlib.h>
 #include <string.h>
 
+#include <ext/pcre/php_pcre.h>
```
Comment (bwoebi) [inline ext/php7/serializer.c:13]:
```suggestion
```
Seems like a leftover?

File: ext/php7/serializer.c (hunk 2 above)
Comment (bwoebi) [reply ext/php7/serializer.c:558]: Right, I was confused with the different names, but yes by default the regex should apply too.

**Review by bwoebi — APPROVED** [review]
Looks good now :-)

**Pull request merged**